### ISSUE TITLE
global: simplify development 

### ISSUE THREAD

**Issue by lnielsen** [inveniosoftware/invenio-cli#170]
Following is a quick brainstorming on commands that potentially could be useful for making it easier to develop with RDM:

Boostrap and install RDM locally:
```
invenio-cli init rdm
cd ~/src/my-site
invenio-cli install --pre
```

Install a Python and JS module for development:
```
invenio-cli module-install -e ~/src/invenio-records-resources
invenio-cli js-install -e ~/src/react-invenio-forms
```

Run:
```
invenio-cli services
invenio-cli run
```

Watch JS changes if needed:
```
invenio-cli watch ~/src/react-invenio-forms
```

Alternative assets commands:
```
invenio-cli assets clean
invenio-cli assets buildall
```

**Comment by fenekku** [comment]
I was thinking about the extensions (https://inveniordm.docs.cern.ch/extensions/) and how the install commands actually fall in that category of functionality.

So having 

`invenio-cli ext init`  To create your own.
`invenio-cli ext js-install ...`  To install
`invenio-cli ext module-install ...` 

could be an idea..

**Comment by lnielsen** [comment]
What should ``invenio-cli ext module-install <PATH>`` do?

```
pipenv run pip install -e <PATH>
```

Should it also run update?
Do we need to write something to the ``.invenio`` file?

**Comment by lnielsen** [comment]
In #173 I've added the commands:

```
invenio assets update --force --development
invenio assets watch
```

The ``--development`` or ``--production`` will respectively set FLASK_ENV=development/production to ensure we either copy or symlinkfiles.
The ``--force`` replaces ``--instal-js/--no-install-js``. Force will clean the folder and install node modules. If not, only a invenio webpack create/build is run (this is needed to wipe assets if e.g. it was created with production and you now want development mode).


For the React module development I'm thinking instead of ``invenio-cli ext js-install ~/src/react-invenio-deposit``, we could have something like:

```
invenio-cli assets watch-module ~/src/react-invenio-deposit
```

This would:

- Install and build module: ``cd ~/src/react-invenio-deposit; npm install; npm run build``
- Link module ``cd ~/src/react-invenio-deposit; npm run link-dist``
- Link project ``cd <path-to-your-virtualenv>/var/instance/assets/node_modules; npm link react-invenio-deposit``
- Start watching changes: ``cd ~/src/react-invenio-deposit; npm run watch``

You will still need afterwards to run:

```
invenio-cli assets watch
```

**Comment by lnielsen** [comment]
A full dev flow would then look like:

```
invenio-cli init rdm -c master
invenio-cli install --pre
invenio-cli ext module-install ~/src/invenio-rdm-records ~/src/invenio-app-rdm
invenio-cli services
invenio-cli demo --local
invenio-cli asset update -f -d
invenio-cli assets watch-module ~/src/react-invenio-deposit
invenio-cli assets watch
invenio-cli run
```

**Comment by lnielsen** [comment]
Now implemented in #173

**Comment by zzacharo** [comment]
@lnielsen for this command `invenio-cli assets watch-module ~/src/react-invenio-deposit` I would add an option to skip the relinking part in case you just want to watch the module i.e `npm run watch` equivalent, assuming you had linked it and now you just want to watch it again :)
